sql/python/sqlflow_submitter/db.py

```diff
@@ -55,6 +56,64 @@ def execute(driver, conn, statement):
     return field_names, field_columns
 
 
+def db_generator(driver, conn, statement,
+                 feature_column_names, label_column_name,
+                 column_name_to_type, fetch_size=128):
+    def reader():
+        cursor = conn.cursor()
+        cursor.execute(statement)
+        if driver == "hive":
+            field_names = None if cursor.description is None \
+                else [i[0][i[0].find('.') + 1:] for i in cursor.description]
+        else:
+            field_names = None if cursor.description is None \
+                else [i[0] for i in cursor.description]
+        label_idx = field_names.index(label_column_name)
+
+        rows = cursor.fetchmany(fetch_size)
+        while len(rows) > 0:
+            for row in rows:
+                label = row[label_idx]
+                features = dict()
+                for name in feature_column_names:
+                    if column_name_to_type[name] == "categorical_column_with_identity":
+                        cell = np.array([int(v) for v in row[field_names.index(name)].split(",")])
```

Comment:
Maybe can we use `np.fromstring` instead to make it efficiency?

Reply:
Good point! Wil do that!

Reply:
I'll do this in next PR so that it's cleaner